packages/commons/engine_util.js

```diff
@@ -590,11 +588,7 @@ function extractJSONPath(doc, expr) {
     return '';
   }
 
-  if (results.length > 1) {
-    return results[randomInt(0, results.length - 1)];
-  } else {
-    return results[0];
-  }
+  return results;
```

Comment:
This is a breaking change to current behavior (we try really hard to not break backwards-compatibility). Perhaps the behavior can be configurable via a parameter to the `json` attribute? e.g.

```yaml
capture:
  - json: '$.results[*]'
    as: results
    multiple: true # return all matches
```

Reply:
Please see the changes I have made